components/Problemset/index.tsx

```diff
@@ -145,6 +147,14 @@ export const Problemset = ({ problems }: ProblemsetProps) => {
                       return false;
                     }
                   }
+                  if (allowedQuestionNumbers) {
+                    if (problem.question === undefined) {
+                      return true;
+                    }
```

Comment:
Filtering by tags and/or difficulty don't work on BIO2 problems because you've returned true here.